src/angular-local-storage.js

```diff
@@ -311,9 +313,23 @@ angular
             }
             if (!!key) {
               var cookiePath = "; path=" + cookie.path;
-              if(cookie.domain){
+              if (cookie.domain) {
                 cookieDomain = "; domain=" + cookie.domain;
               }
+              /* Providing the secure parameter always takes precedence over config 
+               * (allows developer to mix and match secure + non-secure) */
+              if (isDefined(secure) && secure !== null) {
```

Comment:
Why not replace this nested `ifs` for:
```javascript
if (typeof secure === 'boolean' && secure) {
  cookieDomain += '; secure';
}
```

I think it makes the code more readable.